scipy/sparse/linalg/eigen/arpack/arpack.py

```diff
@@ -1822,8 +1827,26 @@ def matvec_XH_X(x):
 
     # Get a low rank approximation of the implicitly defined gramian matrix.
     # This is not a stable way to approach the problem.
-    eigvals, eigvec = eigsh(XH_X, k=k, tol=tol ** 2, maxiter=maxiter,
-                                  ncv=ncv, which=which, v0=v0)
+    if solver == 'lobpcg':
+
+        if which == 'LM':
+            largest = True
+        elif which == 'SM':
+            largest = False
+        else:
+            raise ValueError("which must be either 'LM' or 'SM'.")
+
+        if k == 1 and v0 is not None:
+            X = np.reshape(v0, (-1, 1))
+        else:
+            X = np.random.RandomState(52).randn(min(A.shape), k)
+
+        eigvals, eigvec = lobpcg(XH_X, X, tol=tol ** 2, maxiter=maxiter,
```

Comment:
Is there any reason why we are squaring the tolerances? This should be at least documented, (personal opinion, shouldn't be done at all).

Reply:
I just kept it the same as in eigsh below...

Reply:
Please advise?

Reply:
No change, mark as resolved - please see similar one below